src/input/input_json.rs

```diff
@@ -195,6 +195,22 @@ impl<'a> Input<'a> for JsonInput {
             _ => err_val_error!(input_value = self.as_error_value(), kind = ErrorKind::TupleType),
         }
     }
+
+    fn strict_timedelta(&self) -> ValResult<super::EitherTimedelta> {
+        match self {
+            JsonInput::String(v) => bytes_as_timedelta(self, v.as_bytes()),
+            _ => err_val_error!(input_value = self.as_error_value(), kind = ErrorKind::TimedeltaType),
+        }
+    }
+
+    fn lax_timedelta(&self) -> ValResult<super::EitherTimedelta> {
+        match self {
+            JsonInput::String(v) => bytes_as_timedelta(self, v.as_bytes()),
+            JsonInput::Int(v) => float_as_timedelta(*v as f64),
```

Comment:
humm, this will loose accuracy. How big does `v` need to be before you get a different result?

I think if it's less than 1000 years, we need a separate result for `int_as_timedelta`.